src/ocamldep.ml

```diff
@@ -58,7 +58,44 @@ module Dep_graphs = struct
     Ml_kind.Dict.make_both (Dep_graph.dummy m)
 end
 
-let parse_deps ~dir ~file ~(unit : Module.t)
+let parse_module_names ~dir ~(unit : Module.t) ~modules
+      ~alias_module ~lib_interface_module ~check words =
+  let deps =
+    words
+    |> List.filter_map ~f:(fun m ->
+      let m = Module.Name.of_string m in
+      if m = unit.name then
+        None
+      else
+        Module.Name.Map.find modules m)
+  in
+  if check then
```

Comment:
The rest of this function (the check and addition the alias_module) should go in `parse_deps`. We only need to do these operations once when reading the output of ocamldep.

Reply:
Sure! I wasn't sure if the `alias_module` part was relevant.